src/conduits/RwaOutputConduit3.sol

```diff
@@ -0,0 +1,380 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+pragma solidity 0.6.12;
+
+import {DaiAbstract} from "dss-interfaces/dss/DaiAbstract.sol";
+import {PsmAbstract} from "dss-interfaces/dss/PsmAbstract.sol";
+import {GemAbstract} from "dss-interfaces/ERC/GemAbstract.sol";
+import {GemJoinAbstract} from "dss-interfaces/dss/GemJoinAbstract.sol";
+
+/**
+ * @author Lev Livnev <lev@liv.nev.org.uk>
+ * @author Nazar Duchak <nazar@clio.finance>
+ * @title An Output Conduit for real-world assets (RWA).
+ * @dev This contract differs from the original [RwaOutputConduit](https://github.com/makerdao/MIP21-RWA-Example/blob/fce06885ff89d10bf630710d4f6089c5bba94b4d/src/RwaConduit.sol#L41-L118):
+ *  - The caller of `push()` is not required to hold MakerDAO governance tokens.
+ *  - The `push()` method is permissioned.
+ *  - `push()` permissions are managed by `mate()`/`hate()` methods.
+ *  - `pick` whitelist is managed by `kiss() / diss()` methods.
+ *  - Requires a PSM address in the constructor.
+ *  - `pick` can be called to set the `to` address. Eligible `to` addresses should be whitelisted by an admin through `kiss`.
+ *  - The `push()` method swaps DAI to GEM using PSM and set `to` to zero address.
+ *  - The `push()` method with `amount` argument swaps specified amount of DAI to GEM using PSM and set `to` to zero address.
+ *  - The `quit` method allows moving outstanding DAI balance to `quitTo`. It can be called only by `mate`d addresses.
+ *  - The `quit` method with `amount` argument allows moving specified amount of DAI balance to `quitTo`
+ *  - The `file` method allows updating `quitTo` addresses. It can be called only by the admin.
+ */
+contract RwaOutputConduit3 {
+    /// @notice PSM GEM token contract address.
+    GemAbstract public immutable gem;
+    /// @notice PSM contract address.
+    PsmAbstract public immutable psm;
+    /// @dev DAI/GEM decimal difference.
+    uint256 private immutable to18ConvertionFactor;
+
+    /// @notice Addresses with admin access on this contract. `wards[usr]`
+    mapping(address => uint256) public wards;
+    /// @notice Addresses with operator access on this contract. `can[usr]`
+    mapping(address => uint256) public can;
+
+    /// @dev This is declared here so the storage layout lines up with RwaOutputConduit.
+    address private __unused_gov;
+    /// @notice Dai token contract address.
+    DaiAbstract public dai;
+    /// @notice Dai Recipient address.
+    address public to;
+
+    /// @notice Whitelist for addresses which can be picked. `bud[who]`
+    mapping(address => uint256) public bud;
+    /// @notice Addresses with push access on this contract. `may[usr]`
+    mapping(address => uint256) public may;
+
+    /// @notice Exit address
```

Comment:
mmm this explanation doesn't help much, can further it a bit, especially since we have quit and quitGem now, which can cause confusion

Reply:
done